File: week2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

//prototypes
int count_letters(string s);
int count_words(string s);
int count_sentances(string s);

int main(void)
{
    //getting a string
    string str = get_string("Text: ");

    //getting letters, words, and sentances with functions
    float L = (count_letters(str) / (float)count_words(str)) * 100;
    float S = (count_sentances(str) / (float)count_words(str)) * 100;

    //computing and rounding
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = (int) round(index);

    //printing appropriate grade
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

int count_letters(string s)
{
    //function to find letters
    int letters = 0;
    for (int i = 0; i < strlen(s); i++)
    {
        if (isalnum(s[i]) && !ispunct(s[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string s)
{
    //function to find words
    int words = 1;
    for (int i = 0; i < strlen(s); i++)
    {
        if (isspace(s[i]) && !isspace(s[i + 1]))
        {
            words++;
        }
    }
    return words;
}

int count_sentances(string s)
{
    //function to find sentances
    int sentances = 0;
    for (int i = 0; i < strlen(s); i++)
    {
        if (s[i] == '.' || s[i] == '!' || s[i] == '?')
        {
            sentances++;
        }
    }
    return sentances;
}

File: week6/sentimental-readability/readability.py
# import packages
import cs50
import string


# main function
def main():
    # prompting user for input
    sentence = cs50.get_string("Input: ")

    # computing L and S, then index using specific formula
    L = count_letters(sentence) / count_words(sentence) * 100
    S = count_sentences(sentence) / count_words(sentence) * 100
    index = 0.0588 * L - 0.296 * S - 15.8

    # rounding index to nearest int to get grade
    grade = round(index)

    # printing grade based on value
    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


# function to count letters
def count_letters(s):
    letters = 0
    for i in s:
        if i.isalnum():  # checking if letter alpha-numberic
            if i not in string.punctuation:  # checking if its not a punctuation
                letters += 1
    return letters


# function to count words
def count_words(s):
    words = 1
    for i in range(len(s)):
        # checking if its a space and next is not space
        if s[i].isspace() and (s[i + 1].isspace() != True):
            words += 1
    return words


# function to count sentences
def count_sentences(s):
    sentences = 0
    for i in s:
        if i == "." or i == "!" or i == "?":
            sentences += 1
    return sentences


main()
